x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationTemplate.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.search;
+
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.script.Script;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.optionalConstructorArg;
+
+/**
+ * Search template included in a {@link SearchApplication}. It will be used for searching using the
+ * {@link org.elasticsearch.xpack.application.search.action.QuerySearchApplicationAction}, overriding the parameters included on it via
+ * {@link SearchApplicationQueryParams}
+ */
+public class SearchApplicationTemplate implements ToXContentObject, Writeable {
+    private final Script script;
+
+    public SearchApplicationTemplate(StreamInput in) throws IOException {
+        this.script = in.readOptionalWriteable(Script::new);
+    }
+
+    public SearchApplicationTemplate(Script script) {
+        this.script = script;
```

Comment:
Should we validate that the script language is correctly set to `mustache`?

Reply:
Added 👍